### PR TITLE
helm: Add support for separate Postgres/Mongo listeners in teleport-cluster chart

### PR DESCRIPTION
**Pull request by webvictim**
This PR implements support for separating Postgres and/or Mongo listeners out to a separate port and configuring the `LoadBalancer` appropriately. This is useful when using ACM to terminate traffic in front of the load balancer, as multiplexing on the same port will no longer work.

It also adds a special case to `teleport-cluster` NOTES.txt when using AWS ACM, so we don't show an error when `cert-manager` isn't enabled but ACM is being used.

Backports required:
- `branch/v8`
- `branch/v9`

Linked to #10857

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Comment by webvictim** [comment]
@zmb3 The scope of the PR has changed a bit now with a couple of new values, so would appreciate a re-review.

File: examples/chart/teleport-cluster/tests/config_test.yaml [hunk 1]
```diff
@@ -12,16 +12,6 @@ tests:
...
-  - it: matches snapshot for acme-off.yaml
-    values:
-      - ../.lint/acme-off.yaml
-    asserts:
-      - hasDocuments:
-          count: 1
-      - isKind:
-          of: ConfigMap
-      - matchSnapshot: {}
-
```
Comment (webvictim) [inline examples/chart/teleport-cluster/tests/config_test.yaml:24]: This test was accidentally duplicated twice in the initial commit.

**Comment by webvictim** [comment]
@zmb3 This one has unit tests now... 😁

**Review by zmb3 — APPROVED** [review]

**Review by r0mant — APPROVED** [review]
@webvictim Should we add this to the Helm chart docs too in this PR?

**Comment by webvictim** [comment]
@r0mant Yes, good point. I will add this to the docs, merge and backport tomorrow.

**Comment by webvictim** [comment]
Also adds a quick fix for #11411

**Pull request merged**